pkg/test/utils.go

```diff
@@ -65,15 +66,35 @@ func MustReadFile(t *testing.T, p string) []byte {
 	return b
 }
 
+func extractEventsWithURLPrefix(allEvents, urlPrefix string) string {
+	eventStrings := make([]string, 0)
+	events := strings.Split(allEvents, "---")
+	for _, event := range events {
+		processed := strings.TrimSpace(event)
+		for _, m := range []string{"GET", "POST", "PUT", "PATCH", "DELETE"} {
+			processed = strings.TrimPrefix(processed, m)
+			processed = strings.TrimSpace(processed)
+		}
+		if strings.HasPrefix(processed, urlPrefix) {
+			if len(eventStrings) == 0 {
+				event = strings.TrimPrefix(event, "\n\n")
+			}
+			eventStrings = append(eventStrings, event)
+		}
+	}
+	return strings.Join(eventStrings, "---")
+}
+
 // CompareGoldenFile performs a file comparison for a golden test.
-func CompareGoldenFile(t *testing.T, p string, got string, normalizers ...func(s string) string) {
+func CompareGoldenFile(t *testing.T, path, got string, normalizers ...func(s string) string) {
 	writeGoldenOutput := os.Getenv("WRITE_GOLDEN_OUTPUT") != ""
 
 	for _, normalizer := range normalizers {
 		got = normalizer(got)
 	}
+	gotForComp := got
```

Comment:
Nit: `gotForComparison`.  (And maybe easier to reuse `got`, because comparing `got` and `want` is a well-known pattern.  So easier to stash the `got` value e.g. `fullGot := got`)

Reply:
Yep, using `fullGot` sounds like a cleaner and simpler option. Will update it!